bin/billing/billing.php

```diff
@@ -0,0 +1,281 @@
+
+			// get first letter of type
+			$call_type = strtolower($options['type'][0]);
+
+			// no payments for incoming call else calculate cost for call
+			$call_cost = getCost($options['caller'], $options['callee'], $customer['tariffid']);
+			$price = ($call_type == 'o') ? round(ceil($options['calltime']/$call_cost['unitSize']) * $call_cost['costPerUnit'], 5) : 0;
+
+			// insert cdr record to database
+			$query = sprintf("INSERT INTO
+										 voip_cdr (caller, callee, call_start_time, time_start_to_end, time_answer_to_end, price, status, type, voipaccountid)
+									 VALUES
+										 ('%s', '%s', %s, %d, %d, %f, '%s', '%s', %d);", $options['caller'], $options['callee'], $options['startcall'], $options['totaltime'], $options['calltime'], $price, strtolower($options['status']), $call_type, $customer['voipaccountid']);
+
+			$DB->Execute($query);
+		} else {
+			$fh = (isset($options['file'])) ? fopen($options['file'], 'r') : fopen('php://stdin', 'r');
+			$customer_list = getCustomerList();
+			$error = array();
+			$i=0;
+
+			while($f_line = fgets($fh)) {
+				// increment file line counter
+				++$i;
+
+				// change line to associative array
+				$cdr = parseRow($f_line);
+
+				// check values of cdr array
+				$cdr_error = validCDR($cdr);
+
+				if ($cdr_error === TRUE) {
+					$tariff_id = $customer_list[$cdr['caller']]['tariffid'];
+
+					//include customer tariff
+					if (!isset($tariffs[$tariff_id])) {
+						include_tariff($tariff_id);
+
+						if (!isset($tariffs[$tariff_id])) {
+							$error['errors'][] = array('line'=>$i, 'line_content'=>$f_line, 'error'=>'Cant find tariff ' . $tariff_id . ' in tariff files.');
+							continue;
+						}
+					}
+
+					// get first letter of type
+					$call_type = strtolower($cdr['call_type'][0]);
+
+					// generate unix timestamp
+					$call_start = mktime($cdr['call_start_hour'], $cdr['call_start_min'], $cdr['call_start_sec'], $cdr['call_start_month'], $cdr['call_start_day'], $cdr['call_start_year']);
+
+					// no payments for incoming call else calculate cost for call
+					$call_cost = getCost($cdr['caller'], $cdr['callee'], $tariff_id);
+					$price = ($call_type == 'o') ? round(ceil($cdr['time_answer_to_end']/$call_cost['unitSize']) * $call_cost['costPerUnit'], 5) : 0;
+
+					// insert cdr record to database
+					$query = sprintf("INSERT INTO
+												voip_cdr (caller, callee, call_start_time, time_start_to_end, time_answer_to_end, price, status, type, voipaccountid)
+											VALUES
+												('%s', '%s', %d, %d, %d, %f, '%s', '%s', %d);", $cdr['caller'], $cdr['callee'], $call_start, $cdr['time_start_to_end'], $cdr['time_answer_to_end'], $price, strtolower($cdr['call_status']), $call_type, $customer_list[$cdr['caller']]['voipaccountid']);
+
+					$DB->Execute($query);
+				} else {
+					$error['errors'][] = array('line'=>$i, 'line_content'=>$f_line, 'error'=>$cdr_error);
+					continue;
+				}
+			}
+
+			fclose($fh);
+		}
+	break;
+}
+
+?>
```

Comment:
Chyba brak nowego wiersza na końcu pliku - i to może w wielu miejscach generować ostrzeżenia!